pulsar-client/src/main/java/org/apache/pulsar/client/impl/BinaryProtoLookupService.java

```diff
@@ -103,6 +127,15 @@ public BinaryProtoLookupService(PulsarClientImpl client,
                 histo.withAttributes(Attributes.builder().put("pulsar.lookup.type", "metadata").build());
         histoGetSchema = histo.withAttributes(Attributes.builder().put("pulsar.lookup.type", "schema").build());
         histoListTopics = histo.withAttributes(Attributes.builder().put("pulsar.lookup.type", "list-topics").build());
+
+        if (lookupPinnedExecutor == null) {
+            this.createdLookupPinnedExecutor = false;
```

Comment:
It looks like the variable meaning is currently inverse. I'd assume it to be set to `true` in this case.

Reply:
Done.